public/script.js

```diff
@@ -1,216 +1,273 @@
-var CONSTANTS;
-var STATE = {
-    stop: true,
-    rate: 100,
-    current: moment().year(1990).month(0).day(1),
-    start: moment().year(1990).month(0).day(1),
-    end: moment().year(2018).month(11).day(31),
-    extCon: "extent",
-    norSouth: "n",
-    dateLoopStyle: "daily",
-    monthLoop: 0
-}
-var map;
+let CONSTANTS;
+const STATE = {
+  stop: true,
+  rate: 100,
+  current: moment().year(1990).month(0).day(1),
+  start: moment().year(1990).month(0).day(1),
+  end: moment().year(2020).month(11).day(31),
+  extCon: 'extent',
+  norSouth: 'n',
+  dateLoopStyle: 'daily',
+};
+let map;
+/** Main function run to start animation */
 async function main() {
-    CONSTANTS = await readJSON("constants.json");
-    //Set default settings into the selectors and some other starting values
-    init();
-    
-    $("#animate").click(function () {//When animation button is clicked 
-        if (STATE.stop) {
-            STATE.stop = false;//Start animation
-            $("#animate").html("Stop Animation");
-        }
-        else {
-            STATE.stop = true;
-        $("#animate").html("Start Animation");
-        }
-    });
-
+  CONSTANTS = await readJSON('constants.json');
+  // Set default settings into the selectors and some other starting values
+  init();
 
-    $("#updateParams").click(async function () {
-        updateState();
-    });
+  $('#animate').click(function() {// When animation button is clicked
+    if (STATE.stop) {
+      STATE.stop = false;// Start animation
+      $('#animate').html('Stop Animation');
+    } else {
+      STATE.stop = true;
+      $('#animate').html('Start Animation');
+    }
+  });
 }
+/** Initiaties the input values and the map */
+async function init() {
```

Comment:
Add method to run getCapabilities request and store the resulting dates with data in the global state.